src/components/TodoForm/TodoForm.tsx

```
@@ -0,0 +1,117 @@
+import { useState } from 'react';
+import { getAllUsers, getUserById } from '../../service/UserService';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  setVisibleTodos: (value: React.SetStateAction<Todo[]>) => void;
+  visibleTodos: Todo[];
+};
+export const TodoForm: React.FC<Props> = ({
+  setVisibleTodos,
+  visibleTodos,
+}) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setTitleError] = useState(true);
+  const [isTouched, setIsTouched] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasSelectError, setSelectError] = useState(true);
+
+  const resetForm = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setSelectError(false);
+  };
+
+  const largestId = (array: Todo[]): number =>
+    array.reduce((max, item) => Math.max(max, item.id), 0);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsTouched(true);
+
+    const trimmedTitle = title.trim();
+
+    const hasTitleError = !trimmedTitle;
+    const hasUserError = !userId;
+
+    setTitleError(hasTitleError);
+    setSelectError(hasUserError);
+
+    if (hasTitleError || hasUserError) {
+      return;
+    }
+
+    const newTodo = {
+      id: largestId(visibleTodos) + 1,
+      title: trimmedTitle,
+      completed: false,
+      userId,
+      user: getUserById(userId),
+    };
+
+    setVisibleTodos(prev => [...(prev || []), newTodo]);
```

The use of `prev || []` suggests that the `visibleTodos` prop could be falsy. However, its type is defined as `Todo[]`, which doesn't include `null` or `undefined`. This fallback is unnecessary and could mask a bug. It should be simplified to `setVisibleTodos(prev => [...prev, newTodo])`.